docs: update policy on breaking changes and changelog entry format (#3615)
paritytech/rust-libp2p: docs/release.md

## Development between releases

### Prefer non-breaking changes

We strive to minimize breaking changes for our users.
PRs with breaking changes are not merged immediately but collected in a milestone.
For example: https://github.com/libp2p/rust-libp2p/milestone/6.

Non-breaking changes are typically merged very quickly and often released as patch-releases soon after.

### Make changelog entries

Every crate that we publish on `crates.io` has a `CHANGELOG.md` file.
Substantial PRs should add an entry to each crate they modify.
The next unreleased version is tagged with ` - unreleased`, for example: `0.17.0 - unreleased`.

> In case there isn't a version if an ` - unreleased` postfix yet, add one for the next version.

Author: ```suggestion
In case there isn't a version with an ` - unreleased` postfix yet, add one for the next version.
```

> The version number depends on the impact of your change (breaking vs non-breaking, see above).

Author: ```suggestion
The next version number depends on the impact of your change (breaking vs non-breaking, see above).
```

If you are making a non-breaking change, please also bump the version number in the `Cargo.toml` manifest.
In case another crate _depends_ on the new features or APIs that you are adding, you may need to bump the dependency declaration in that `Cargo.toml` file as well.
Don't worry if you don't get this one right, we will flag it in the PR if necessary :)

For breaking changes, a changelog entry itself is sufficient.

> Bumping the version in the `Cargo.toml` file itself as well leads to merge conflicts once we decide to merge them.

Author: ```suggestion
Bumping the version in the `Cargo.toml` file would lead to many merge conflicts once we decide to merge them.
Hence, we are going to bump those versions once we work through the milestone that collects the breaking changes.
```
